src/shared/concentrators/AMMConcentratorBase.sol

```diff
@@ -266,7 +265,29 @@ abstract contract AMMConcentratorBase is ReentrancyGuard, ERC4626 {
     /// @param _receiver - The receiver of minted shares.
     /// @param _minAmount - The minimum amount of assets (LP tokens) to receive.
     /// @return _shares - The amount of shares minted.
-    function depositSingleUnderlying(uint256 _underlyingAmount, address _underlyingAsset, address _receiver, uint256 _minAmount) external payable virtual nonReentrant returns (uint256 _shares) {}
+    // slither-disable-next-line reentrancy-no-eth
+    function depositSingleUnderlying(uint256 _underlyingAmount, address _underlyingAsset, address _receiver, uint256 _minAmount) external payable nonReentrant returns (uint256 _shares) {
+        if (!_isUnderlyingAsset(_underlyingAsset)) revert NotUnderlyingAsset();
+        if (!(_underlyingAmount > 0)) revert ZeroAmount();
+        
+        _updateRewards(msg.sender);
```

Comment:
Is there a reason to "update rewards" for the user this early? Is there any implication to doing it here vs at the end? Any attack vectors here that we have to pay attention to?

Reply:
Timing doesnt matter in this instance. This function violates the check-effect-interact pattern (which prevents against reentrancy attacks), but is using OZ's `nonReentrant` modifier.

We could use check-effect-interact pattern in this function, but we would need to assume that all the slippage the user is agreeing to have will actually happen (basically using `_minAmount` as `_assets`).